src/generator/generator_visitor.ts

```diff
@@ -321,21 +371,44 @@ class GeneratorVisitor extends Visitor {
 
     private handleShorthandAssignment(node: AssignmentExpression) {
         this.visit(node.left);
-        const currentValue = this.popExpression();
-        const assignedValue = this.popExpression();
+        let currentValue = this.popExpression();
+        let assignedValue = this.popExpression();
+
+        if (!isIdentifier(node.left) && !isMemberExpression(node.left)) {
+            throw new Error('Shorthand assignment only allowed on an identifier or an array member');
```

Comment:
No it is never tested, because our language specification doesn't allow any other expressions on the left side of a shorthand assignment. Therefore it is almost impossible to test and to make sure, that the error was thrown at this location.

This check is only there, because `node.left` is used as a parameter of the `getExpressionType` call.